src/main/java/macrobase/analysis/transform/BatchSWTransform.java

```diff
@@ -0,0 +1,60 @@
+package macrobase.analysis.transform;
+
+import macrobase.analysis.pipeline.stream.MBStream;
+import macrobase.analysis.transform.aggregate.BatchWindowAggregate;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+
+import java.util.List;
+
+public class BatchSWTransform extends SlidingWindowTransform {
+    private BatchWindowAggregate windowAggregate;
+
+    public BatchSWTransform(MacroBaseConf conf, int windowSize) throws ConfigurationException {
```

Comment:
I was thinking the opposite: what about setting window size in conf?

If we want different window sizes for different operators, we can change the conf instance.

The goal of the current conf design is to avoid passing in all params at construction and instead going through conf. This simplifies a few things, including reasoning about default args.